linux_service/moneo_prestart.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+MONEO_PATH=$1
+EXE_TYPE=$2
+
+# check MONEO_PATH variable is set if not set to default
+if [[ -z "$MONEO_PATH" ]];
+then
+    MONEO_PATH=/opt/azurehpc/tools/Moneo
+    echo 'default Moneo path used'
+fi
+echo "Moneo path=$MONEO_PATH"
+
+if [[ -z "$EXE_TYPE" ]];
+then
+    echo 'Error: No executable passed in. Exiting prestart script.'
+    exit 1
+fi
+
+# check that the path provided exists
+if [[ ! -d "$MONEO_PATH" ]];
+then
+    echo "Error: Moneo path does not exist. Please install Moneo or provide path to this script. Exiting prestart script"
+    exit 1
+fi
+
+# check/create the working director exists
+mkdir -p /tmp/moneo-worker/exporters
+mkdir -p /tmp/moneo-worker/publisher
+
+# copy exporters or publisher
+if [[ "metrics_publisher.py" == "$EXE_TYPE"  ]];
+then
+    if [[ ! -e "$MONEO_PATH/src/worker/publisher/$EXE_TYPE" ]];
+    then
+        echo "$MONEO_PATH/src/worker/publisher/$EXE_TYPE Does not exist"
+        exit 1
+    fi
+    cp $MONEO_PATH/src/worker/publisher/$EXE_TYPE  /tmp/moneo-worker/publisher/
+else
+    if [[ ! -e "$MONEO_PATH/src/worker/exporters/$EXE_TYPE" ]];
```

Comment:
Can we pull this existence check out of the if statement at 32, since we have to check for it in each case?

Reply:
Isn't line 41 specific to $MONEO_PATH/src/worker/**exporters** so if it's moved out of the line 32 _if_ there would need to be a check that EXE_TYPE is not publisher